diff --git a/_lib/vite/utils/search.js b/_lib/vite/utils/search.js
index 2f120d050574cecfae452d746a025c89e31652f9..b580cd4364556d16b40828b894ba536b944f7f2a 100644
--- a/_lib/vite/utils/search.js
+++ b/_lib/vite/utils/search.js
@@ -27,7 +27,11 @@ export async function buildIndex({ baseDir, cacheDir }) {
         const pageCache = cache.get(key) ?? {};
         if (pageCache.mdx === mdx)
             return pageCache.document;
-        const { html, frontmatter } = await processMdx(pagePath, mdx, { rehypePlugins });
+        
+        const mdxNoImports = mdx.replace(/^(import|export).*/gm, '');
+        const mdxNoComponents = mdxNoImports.replace(/<\/?\s*[A-Z][^>]*>/g, '');
+        const { html, frontmatter } = await processMdx(pagePath, mdxNoComponents, { rehypePlugins });
+
         if (frontmatter.searchable === false) {
             cache.set(key, { mdx, document: [] });
             return [];
